### ISSUE TITLE
VLAN and VLAN group parent location inheritance

### ISSUE THREAD

**Issue by ejfuria** [nautobot/nautobot#5050]
### Environment
* Nautobot version: 2.1.0 (docker tag networktocode/nautobot:2.1-py3.10)
* Python version: 3.10
* Database platform: postgres
* Database version: 14

### Steps to Reproduce
1.  Create a building (location) <building-name>
2.  Create an new location <room-name> with a building as the parent location
3.  Create a VLAN or a VLAN group with a location that is a building (<building-name>)
4.  Create a device with a location of <room-name>
5.  Create in interface with "802.1Q Switching" and assigned a VLAN to the interface 

### Expected Behavior
Expected to be able to select a VLAN that is associated with the parent building

### Observed Behavior
There were no VLAN selection options.    This does work with VLAN groups in Netbox.

**Comment by jifox** [comment]
Hi @ejfuria

Have you checked that the `Content types` field in Location Type `Building` contains `ipam | VLAN group`?

**Comment by ejfuria** [comment]
Yes sir - the **Content types** field in Location Type Building contains ipam | VLAN group and ipam | VLAN.  The same goes for my Location Type "Room"

**Comment by glennmatthews** [comment]
Needs to be addressed as part of our broader story around tree-model data inheritance - see also #4413.

**Comment by cardoe** [comment]
Is this a bit like the need for "logical" containers as well as "physical" containers like #3110 touches on @glennmatthews ?

**Comment by alextremblay** [comment]
Hi @glennmatthews I'd love to get a better perspective on this: to me this seems like a bug.

prior to nautobot 2.0, when `site` field was still around, it was not only possible to model this expected behaviour, it was actually the default! adding vlans to sites, and adding switches to rooms (locations) while also adding them to those same sites, everything more or less worked as expected.

now that site is just another type of location, we have this added complexity that the above behaviour can ONLY be triggered if we abandon rooms and floors and stick all our switches in the same location as our VLANs

To me, the ability to pull VLANs for a device from the device's location's parents doesn't feel like a feature, the lack of such behaviour feels like a bug.

thoughts?

**Comment by jifox** [comment]
Seems to be related: 
https://github.com/nautobot/nautobot/issues/5930

**Comment by cardoe** [comment]
I believe VLANGroups should have a scope that they are bound to Location, Rack, Cluster or ClusterGroup like #5903 mentioned.
